resolwe/flow/managers/base.py

```diff
@@ -53,37 +52,31 @@ def default(self, o):  # pylint: disable=method-hidden
 def dependency_status(data):
     """Return abstracted satus of dependencies.
 
-    STATUS_ERROR .. one dependency has error status
+    STATUS_ERROR .. one dependency has error status or was deleted
     STATUS_DONE .. all dependencies have done status
     None .. other
 
     """
-    for field_schema, fields in iterate_fields(data.input, data.process.input_schema):
-        if (field_schema['type'].lower().startswith('data:') or
-                field_schema['type'].lower().startswith('list:data:')):
-            name = field_schema['name']
-            value = fields[name]
+    parents_status = set(
```

Comment:
I would rename this to `parent_statuses` or `parents_statuses` since the lines below are a bit confusing when you check if `len(parents_status)` is `0` or `1` and the name implies there is one status in the variable.